Cabal/doc/developing-packages.rst

```diff
@@ -2063,21 +2063,31 @@ system-dependent values for these fields.
 .. pkg-field:: build-tool-depends: package:executable list
     :since: 2.0
 
-    A list of Haskell programs needed to build this component.
+    A list of Haskell executables needed to build this component. Executables are provided during the whole duration of the build,
+    so this field can be used for executables needed during :pkg-section:`test-suite` as well.
```

Comment:
Yes :) Anything to clarify here?

Reply:
@Ericson2314 btw, assuming we get `run-tool-depends`  at some point; we'd still need to emulate the current `build-tools-depends` which conflatedly implies the future `run-tool-depends` w/ `cabal-version:2.*` packages; so at that point, @domenkozar wording will still be valid for packages using the respective `cabal-version:`-values which still had these semantics.

Reply:
Btw, that sentence implies that running the test-suite is considered part of the build-phase; I think it's worth spelling this out as otherwise it might not be obvious that the respective `$PATH`-augmentation will also be in effect during the invocation of the test-suite (via e.g. `cabal new-test`)

Reply:
@quasicomputational you picked up on my subtle hint... ;-)

I consider this a bit of a bug, and I seem to recall we already talked about this in another issue; i.e. ideally

- `cabal new-run test:x` behaves close to `cabal new-test x` 
- `cabal new-run bench:x` behaves close to `cabal new-bench x` 

as I'd consider it merely a different interface to invoking the test-suites; but the $PATH setup ought to be compatible.

Reply:
Right insofar that there's a bug, I consider the `new-run` behavior correct :D.